lib/pure/times.nim

```diff
@@ -1042,35 +1226,121 @@ proc countYearsAndDays*(daySpan: int): tuple[years: int, days: int] =
   result.years = days div 365
   result.days = days mod 365
 
-proc getDayOfWeek*(day, month, year: int): WeekDay =
+proc getDayOfYear*(monthday: MonthdayRange, month: Month, year: int): YeardayRange =
+  ## Returns the day of the year.
+  ## Equivalent with ``initDateTime(day, month, year).yearday``.
+  const daysUntilMonth : array[Month, int] = [0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334]
+  const daysUntilMonthLeap : array[Month, int] = [0, 31, 60, 90, 120, 151, 181, 212, 243, 273, 304, 334]
+
+  if isLeapYear(year):
+    result = daysUntilMonthLeap[month] + monthday - 1
+  else:
+    result = daysUntilMonth[month] + monthday - 1
+
+proc getDayOfWeek*(monthday: MonthdayRange, month: Month, year: int): WeekDay =
   ## Returns the day of the week enum from day, month and year.
-  # Day & month start from one.
+  ## Equivalent with ``initDateTime(day, month, year).weekday``.
   let
-    a = (14 - month) div 12
+    ordinalMonth = ord(month)
+    a = (14 - ordinalMonth) div 12
     y = year - a
-    m = month + (12*a) - 2
-    d = (day + y + (y div 4) - (y div 100) + (y div 400) + (31*m) div 12) mod 7
+    m = ordinalMonth + (12*a) - 2
+    d = (monthday + y + (y div 4) - (y div 100) + (y div 400) + (31*m) div 12) mod 7
   # The value of d is 0 for a Sunday, 1 for a Monday, 2 for a Tuesday, etc.
   # so we must correct for the WeekDay type.
-  if d == 0: return dSun
-  result = (d-1).WeekDay
+  result = if d == 0: dSun else: WeekDay(d - 1)
 
-proc getDayOfWeekJulian*(day, month, year: int): WeekDay =
-  ## Returns the day of the week enum from day, month and year,
-  ## according to the Julian calendar.
-  # Day & month start from one.
-  let
-    a = (14 - month) div 12
-    y = year - a
-    m = month + (12*a) - 2
-    d = (5 + day + y + (y div 4) + (31*m) div 12) mod 7
-  result = d.WeekDay
+proc toTimeInterval*(time: Time): TimeInterval =
+  ## Converts a Time to a TimeInterval.
+  ##
+  ## To be used when diffing times.
+  ##
+  ## .. code-block:: nim
+  ##     let a = fromSeconds(1_000_000_000)
+  ##     let b = fromSeconds(1_500_000_000)
+  ##     echo a, " ", b  # real dates
+  ##     echo a.toTimeInterval  # meaningless value, don't use it by itself
+  ##     echo b.toTimeInterval - a.toTimeInterval
+  ##     # (milliseconds: 0, seconds: -40, minutes: -6, hours: 1, days: -2, months: -2, years: 16)
+  # Milliseconds not available from Time
+  var dt = time.local
+  initInterval(0, dt.second, dt.minute, dt.hour, dt.weekday.ord, dt.month.ord - 1, dt.year)
```

Comment:
Using `dt.weekday.ord` is simply nonsense here. It should be `dt.monthday`. The result given is necessarily wrong (see my other comment). A datetime is in principle nothing other than a description of a TimeInterval since some arbitrary starting point. The day of week is not a meaningful part of this interval.